Updated loop docs (#838)
ansible/ansible-documentation: docs/docsite/rst/playbook_guide/playbooks_loops.rst

repeating a polling step until a certain result is reached.

.. note::
   * We added ``loop`` in Ansible 2.5. as a simpler way to do loops, but we recommend it for most use cases.
   * We have not deprecated the use of ``with_<lookup>`` - that syntax will still be valid for the foreseeable future.

>    * ``loop`` and ``with_<lookup>`` are mutually exclusive, while it is possible to nest them under ``until``, this affects each loop iteration.

Reviewer: ```suggestion
   * ``loop`` and ``with_<lookup>`` are mutually exclusive. While it is possible to nest them under ``until``, this affects each loop iteration.
```

.. contents::
   :local:

Comparing loops
===============

* The normal use case for ``until`` has to do with tasks that are likely to fail, while ``loop`` and ``with_<lookup>`` are meant for repeating tasks with slight variations.
* The ``loop`` and ``with_<lookup>`` will run the task once per item in the list used as input, while ``until`` will rerun the task until a condition is met.
  For programmers the former are "for loops" and the latter is a "while/until loop".
* The ``with_<lookup>`` keywords rely on :ref:`lookup_plugins` - even  ``items`` is a lookup.
* The ``loop`` keyword is equivalent to ``with_list``, and is the best choice for simple loops.
* The ``loop`` keyword will not accept a string as input, see :ref:`query_vs_lookup`.
* The ``until`` keyword accepts an 'end conditional' (expression that returns ``True`` or ``False``)  that is "implicitly templated" (no need for ``{{ }}``),
  commonly based on the variable you ``register`` for the task.

> * ``loop_control`` affects both ``loop`` and ``with_<lookup>``, but not ``until``, which has it's own companion keywords: ``retries`` and ``delay``.

Reviewer 2: ```suggestion
* ``loop_control`` affects both ``loop`` and ``with_<lookup>``, but not ``until``, which has its own companion keywords: ``retries`` and ``delay``.
```

* Generally speaking, any use of ``with_*`` covered in :ref:`migrating_to_loop` can be updated to use ``loop``.

> * Be careful when changing ``with_items`` to ``loop``, as ``with_items`` performed implicit single-level flattening.

Reviewer 2: ```suggestion
* Be careful when changing ``with_items`` to ``loop``, as ``with_items`` performs implicit single-level flattening.
```

>   You may need to use ``flatten(1)`` with ``loop`` to match the exact outcome. For example, to get the same output as:

Reviewer 2: Perhaps better to include ` |` to make clear this is a filter.

```suggestion
  You may need to use ``| flatten(1)`` with ``loop`` to match the exact outcome. For example, to get the same output as:
```

Stacking loops via include_tasks
--------------------------------
.. versionadded:: 2.1

You can nest two looping tasks using ``include_tasks``. However, by default, Ansible sets the loop variable ``item`` for each loop.
This means the inner, nested loop will overwrite the value of ``item`` from the outer loop.
To avoid this, you can specify the name of the variable for each loop using ``loop_var`` with ``loop_control``.

.. code-block:: yaml+jinja

    # main.yml
    - include_task: inner.yml
      loop:
        - 1
        - 2
        - 3
      loop_control:
        loop_var: outer_item

    # inner.yml
    - name: Print outer and inner items
      ansible.builtin.debug:
        msg: "outer item={{ outer_item }} inner item={{ item }}"
      loop:
        - a
        - b
        - c

.. note:: If Ansible detects that the current loop is using a variable that has already been defined, it will raise an error to fail the task.

Until and loop
--------------

The ``until`` condition will apply per ``item`` of the ``loop``:

.. code-block:: yaml+jinja

    - debug: msg={{item}}
      loop:
        - 1
        - 2
        - 3
      retries: 2
      until: item > 2

> This will make Ansible retry the first 2 items and extre 2 times, then fail the item on the 3rd attempt,

Reviewer: ```suggestion
This will make Ansible retry the first 2 items twice, then fail the item on the 3rd attempt,
```

then succeed at the first attempt on the 3rd item, in the end failing the task as a whole.

> .. code-block:: none

Reviewer 2: > .. code-block:: none

Don't we have a special output type for playbook output?

Author: I have not found it, looked at other examples, that is where i got `none` from
